java/arcs/core/util/TaggedLog.kt

```diff
@@ -11,12 +11,27 @@
 package arcs.core.util
 
 /**
- * Allows the user to create a scoped log utility where everything logged will be prepended with the output of
- * [tagBuilder].
+ * Allows the user to create a scoped log utility where everything logged will be prepended with the
+ * output of [tagBuilder]. An optional suffix can be appended using [withSuffix] when constructing.
  */
 class TaggedLog(private val tagBuilder: () -> String) {
+    private var suffixBuilder: (() -> String)? = null
+
     private fun taggedMessageBuilder(messageBuilder: () -> String): () -> String {
```

Comment:
add a param to this like taggedMessageBuilder(debug: Boolean = false, messageBuilder: () -> String)

Reply:
and change TaggedLog.debug() to pass debug = true when calling the taggedMessageBuilder, then withSuffix can contain PII unsafe stuff.